File: Ejercicio 4/palindromo.js
function esPalindromo(palabra) {
    // Convertir a minúsculas y eliminar espacios
    palabra = palabra.toLowerCase().replace(/\s/g, '');
    // Convertir la palabra en un array, invertirlo y unirlo de nuevo
    const palabraInvertida = palabra.split('').reverse().join('');
    return palabra === palabraInvertida;
}

// Función principal
function main() {
    // Solicitar entrada al usuario
    const palabra = prompt("Introduce una palabra para verificar si es un palíndromo:");
    
    // Verificar si es palíndromo
    if (esPalindromo(palabra)) {
        console.log(`"${palabra}" es un palíndromo!`);
    } else {
        console.log(`"${palabra}" no es un palíndromo.`);
    }
}

// Ejecutar la función principal
main();

File: Ejercicio 4/Palindromo.java
import java.util.Scanner;

public class Palindromo {
    public static boolean esPalindromo(String palabra) {
        // Convertir a minúsculas y eliminar espacios
        palabra = palabra.toLowerCase().replaceAll("\\s+", "");
        
        // Crear un StringBuilder para invertir la palabra
        StringBuilder palabraInvertida = new StringBuilder(palabra).reverse();
        
        // Comparar la palabra original con la invertida
        return palabra.equals(palabraInvertida.toString());
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Solicitar entrada al usuario
        System.out.print("Introduce una palabra para verificar si es un palíndromo: ");
        String palabra = scanner.nextLine();
        
        // Verificar si es palíndromo
        if (esPalindromo(palabra)) {
            System.out.println('"' + palabra + '"' + " es un palíndromo!");
        } else {
            System.out.println('"' + palabra + '"' + " no es un palíndromo.");
        }
        
        scanner.close();
    }
}